python/lib/dependabot/python/language_version_manager.rb

```diff
@@ -23,7 +23,7 @@ def install_required_python
         return if SharedHelpers.run_shell_command("pyenv versions").include?(" #{python_major_minor}.")
 
         SharedHelpers.run_shell_command(
-          "tar xzf /usr/local/.pyenv/#{python_major_minor}.tar.gz -C /usr/local/.pyenv/"
+          "tar xzf /usr/local/.pyenv/versions/#{python_version}.tar.gz -C /usr/local/.pyenv/versions"
```

Comment:
I think this has been brought up in conversation elsewhere, but this change would mean some projects using an older patch version would be run using the latest matching minor version

That should be okay though as long as the patch updates are actually non-breaking...

Reply:
Not sure I follow... I think this only changes an implementation detail of the name of the tar file? The result of the untarring doesn't actually change.

Also, the matching logic for older patch getting coerced to newer patch doesn't happen here, it happens over in:
https://github.com/dependabot/dependabot-core/blob/78bd848e8881fc0bec480ed1ca809b0c40efab5d/python/lib/dependabot/python/language_version_manager.rb#L54-L61

And we already  have been doing that for months.

Reply:
Understood. I guess I had mixed the two up

Reply:
It's super confusing, my memory just happens to be fresh since I rewrote that code a few weeks ago as we could simplify the logic a bit after we dropped python 3.6/3.7...